DIPs/DIPNNNN.md

```diff
@@ -0,0 +1,349 @@
+
+#### With single global namespace
+
+Examples: C, early PHP
+
+Such languages tend to rely on [using obscure names](#using-more-obscure-name)
+that makes symbol clash less likely but in general problem is still present.
+When the problem manifests, situation becomes much worse than in D, because such
+languages lack tools to disambiguate name clash explicitly.
+
+#### With namespaces, allowing non-qualified access
+
+Examples: C++, C#
+
+This category is most simple to D, meaning that language provides tools to
+disambiguate any names but at the same time allows imports with non-qualified
+access to imported symbols.
+
+In D this is default `import` behavior.
+
+In C++ : `using namespace Namespace.Subnamespace`.
+
+In C# : `using Namespace.Subnamespace`
+
+Importance of problem varies depending on idiomatic code style dominant for the
+language and other available tools. For example, C# documentation
+[recommends](https://msdn.microsoft.com/en-us/library/dfb3cx8s.aspx) using
+dedicated namespace access operator `::` instead of uniform `.` to reduce the
+risk of clashes.
+
+However, there are few D specifics that make issues somewhat more important:
+
+- In all mentioned languages but D it is common convention to
+  only use unqualified access for symbols in standard library. In D plain
+  `import` is most widespread way to use all libraries, including dub registry.
+- D runtime features module `object.d` which is implicitly imported everywhere,
+  greatly increasing chance for name clash for any symbol in it.
+
+#### With namespaces, restricted non-qualified access
+
+Examples: Rust, Go
+
+In both Rust and Go normal imports still require using module name for
+qualification. For example, in Rust `use Namespace::Subnamespace::Module` still
+requires to explicitly write `Module::foo()` to call `foo` function.
+
+Both languages provide a way to enable unqualified access (`import . "Module"`
+in Go, `use Namespace::Module::*` in Rust) but this feature is heavily
+discouraged from usage in production projects exactly because of problems
+described in this DIP.
+
+This completely solves the issue described in this DIP but is too different
+to established idiomatic D coding style to be considered.
+
+### Breaking changes / deprecation process
+
+Proposed semantics are purely additive and can't affect existing code. However,
+this DIP expects high chance of introducing bugs during initial implementation
+because symbol resolution in D is non-trivial.
+
+Because of that, intended implementation approach is to initially introduce
+proposed concept as a feature internal to compiler and use for druntime symbols
+only. Otherwise there is a certain chance of releasing feature with bug in
+symbol resolution semantics which will become relied upon and widely used as
+a feature, preventing from easily fixing to act as intended later.
+
+Once implementation is confirmed to be solid, adding actual syntax support is
+much more trivial and less bug-prone task.
```

Comment:
s/much more trivial and less bug-prone task/a trivial task unlikely to cause subsequent bugs/